PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1

```diff
@@ -309,10 +282,7 @@ $CommonFuncBlock = {
 						'Microsoft-Windows-StorageReplica/Performance',
                         'Microsoft-Windows-StorageSpaces-Driver/Performance'
 
-        # Core logs to gather, by explicit names.
-        $LogPatterns += 'System','Application'
-
-        Get-WinEvent -ListLog $LogPatterns -Force -ErrorAction Ignore -WarningAction Ignore |? { $LogToExclude -notcontains $_.LogName } |% {
+        Get-WinEvent -ListLog *  -Force -ErrorAction Ignore -WarningAction Ignore | ? {if ($_.RecordCount) {$_.LogName}} |? { $LogToExclude -notcontains $_.LogName } |% {

```

Comment:
we need to capture the -ListLog objects to a new abc.xml so we can see them for triage/confirmation.
can you check that filter statement? it looks like you're outputting $_.LogName onto the pipe, not the object, so I'm not sure the $LogToExclude check on $_.LogName can work. I would simply write it as |? RecordCount |? ...